### ISSUE TITLE
The `g` goto menu should be case sensitive

### ISSUE THREAD

**Issue by lenormf** [mawww/kakoune#3976]
Hi,

All mappings added to the goto menu (<kbd>g</kbd>) are case insensitive.

I don't think it makes sense, since users could want to declare an alternative mapping to the default ones using <kbd>shift</kbd> (more practical than <kbd>alt</kbd>).

The shift modifier also has a significant meaning in the context of mappings (it generally means "extend"), despite that meaning being superseded by the <kbd>G</kbd> menu.

HTH.

**Comment by Screwtapello** [comment]
I assume they're case-insensitive so that you can select to the beginning of the line with GH. The idea of "hold shift to extend the selection" is easy enough, but for two-key movements like the goto menu, it's not immediately clear whether you need to hold shift for just the first, or for both keys. Plus, if you try to type `Gh` quickly you might wind up typing `GH` accidentally anyway.

All things considered, it's friendlier just to make the goto mode case-insensitive, so that what users type (due to confusion or to accident) Just Works.

**Comment by toddyamakawa** [comment]
I deliberately type `gL` and `gH` all the time and I keep getting surprised when it doesn't highlight my movement. I think `gL` and `gH` are very difficult to accidentally type.

**Comment by mawww** [comment]
@whereswaldon Agreed, the fact that goto mappings are not split between extend/non-extend does not seem like a good thing, at the same time I am not really keen to solve that by introducing goto/goto-extend modes, I'd hope for a more generic solution but I have no design in mind.

**Comment by whereswaldon** [comment]
Would it be sane to ignore case on those mappings if and only if the character used to enter the mode was capitalized? I don't know the logistics required to implement that.
